apis/python/src/tiledbsoma/tiledb_create_options.py

```diff
@@ -14,85 +14,50 @@
-    X_cell_order: str = "row-major"
-    # https://github.com/single-cell-data/TileDB-SOMA/issues/27
-    string_dim_zstd_level: int = 3
-    write_X_chunked: bool = True
-    goal_chunk_nnz: int = 200_000_000
-    # Allows relocatability for local disk / S3, and correct behavior for TileDB Cloud
-    member_uris_are_relative: Optional[bool] = None
-
-    max_thread_pool_workers: int = 8
-
-
-def from_param(pc: Optional[PlatformConfig]) -> "ParamWrapper":
-    """Extracts the TileDB-specific platform config key."""
-    pc = pc or {}
-    return ParamWrapper(pc.get("tiledb", {}))
+DEFAULT_TILE_ORDER = "row-major"
+DEFAULT_CELL_ORDER = "row-major"
+DEFAULT_STRING_DIM_ZSTD_LEVEL = 3
+DEFAULT_WRITE_X_CHUNKED = True
+DEFAULT_GOAL_CHUNK_NNZ = 200_000_000
+DEFAULT_FILTERS = (
+    "DoubleDeltaFilter",
+    "BitWidthReductionFilter",
+    "ZstdFilter",
+)
+DEFAULT_TILE_EXTENT = 2048
+# TODO: pending further work on
+#  https://github.com/single-cell-data/TileDB-SOMA/issues/27
+# DEFAULT_OBS_EXTENT = 256
+# DEFAULT_VAR_EXTENT = 2048
+# DEFAULT_X_CAPACITY = 100000
+# DEFAULT_MAX_THREAD_POOL_WORKERS = 8
 
 
 @dataclass(frozen=True)
-class ParamWrapper(Mapping[str, Any]):
-    """Wrapper for the TileDB key of the ``platform_config`` parameter.
-
-    Provides some methods and proxies to the config dict itself.
-    """
-
-    _config: PlatformConfig
-
-    def create_options(self) -> "CreateWrapper":
-        return CreateWrapper(self.get("create", {}))
-
-    # Mapping implementation:
-
-    def __getitem__(self, it: str) -> Any:
-        return self._config[it]
-
-    def __len__(self) -> int:
-        return len(self._config)
+# TODO: Inherit from SOMA.PlatformConfig instead (which itself is a `
+#  Mapping[str, Any]`)
+class TileDBCreateOptions(Mapping[str, Any]):
 
-    def __iter__(self) -> Iterator[str]:
-        return iter(self._config)
+    StrOrMap = Union[str, Mapping[str, Any]]
 
+    _config: Mapping[str, Any] = field(default_factory=dict)
 
-StrOrMap = Union[str, Mapping[str, Any]]
+    def string_dim_zstd_level(self) -> int:
+        return self.get("string_dim_zstd_level", DEFAULT_STRING_DIM_ZSTD_LEVEL)
 
+    def write_X_chunked(self) -> bool:
+        return self.get("write_X_chunked", DEFAULT_WRITE_X_CHUNKED)
 
-@dataclass(frozen=True)
-class CreateWrapper(Mapping[str, Any]):
-
-    _config: Mapping[str, Any]
+    def goal_chunk_nnz(self) -> int:
+        return self.get("goal_chunk_nnz", DEFAULT_GOAL_CHUNK_NNZ)
 
     def offsets_filters(
         self,
-        default: Sequence[StrOrMap] = (
-            "DoubleDeltaFilter",
-            "BitWidthReductionFilter",
-            "ZstdFilter",
-        ),
+        default: Sequence[StrOrMap] = DEFAULT_FILTERS,
     ) -> Sequence[tiledb.Filter]:
         return _build_filters(self.get("offsets_filters", default))
 
-    def cell_tile_orders(
-        self,
-        default_cell: Optional[str] = "row-major",
-        default_tile: Optional[str] = "row-major",
```

Comment:
The reason I did this here is in case we wanted to use different default cell/tile reading orders in different places. It is entirely possible that we don’t, and removing this is honestly probably fine given that—it can always be added back later.

Reply:
As every `create()` call takes a `TileDbCreateOptions` arg, wouldn't every `TileDbArray` be able to have its own cell/tile order specified?  As shown in the [test](https://github.com/single-cell-data/TileDB-SOMA/blob/cce50edf05f37da122564d92cc08b9dd66baec64/apis/python/tests/test_create_options.py#L47), specifying `cell_order` and `row_order` is still supported.

Reply:
keeping as is for now